### DISCUSSION TITLE
Documentation System & Roadmap

### DISCUSSION THREAD

**Discussion by amcgee** [dhis2/notes#210] [Ideas]
### The documentation system

Hosted at [developers.dhis2.org](https://developers.dhis2.org) (adapted from the [divio documentation system](https://documentation.divio.com/))

Type | Orientation | Goals | Useful while
-- | -- | -- | --
Concept | Understanding | Get a broad overview of the system Understand why design decisions were made | Learning
Tutorial | Learning | Get started with the tools (from zero) | Learning
Guide | Problem | Solve a specific problem (from non-zero) | Working
Reference | Information | Exhaustive API specifications | Working

**Some Guide ideas** (no particular order)

* How to build a library with the Application Platform
* How to fetch data from a DHIS2 instance
* How to test my application for a specific DHIS2 version (with `d2 cluster`)
* How to add a DHIS2 UI Button to my application
* How to add a DHIS2 Org Unit Tree selector to my application
* How to add translation support to an application
* How to add page routing to an application
* How to add global state to an application
* How to use a sidebar layout in my application
* How to save and restore user settings in my application
* How to save and load "visualization" objects in my application
* How to embed a plugin in my application
* How to build a dashboard plugin
* How to write automated tests for my application
* How to submit an application to the DHIS2 App Hub
* How to ensure security and privacy in DHIS2 apps

**Comment by amcgee** [comment]
CC @awgaan for any additional topics I missed

**Comment by amcgee** [comment]
@deboragaleano I'd like to add another important guide topic that we should cover - `How to ensure security and privacy in DHIS2 apps`

This might need to be a whole series

**Reply by deboragaleano** [reply]
Yes, this is important @amcgee ! I'll take note of this 👌
